app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -606,7 +611,7 @@ BrowserWorld::UpdateWidget(int32_t aHandle, const WidgetPlacementPtr& aPlacement
     newWorldWidth = aPlacement->width * kWorldDPIRatio;
   }
 
-  if (newWorldWidth != worldWidth) {
+  if (newWorldWidth != worldWidth || oldWidth != aPlacement->width) {
```

Comment:
Don't we need to check height? What happens if both worldWidth & aPlacement->width are equal, but height changes?

Reply:
That's what I thought but for some reason when scaling in portrait mode what changes is the width not the height. Anyway I've added both checks just in case although the one that makes it work is the width one.